### PR TITLE
Noderesourcetopology- api: Repo Bootstrap

### LINKED ISSUE #1: Create a SECURITY_CONTACTS file.
As per the email sent to kubernetes-dev[1], please create a SECURITY_CONTACTS
file.

The template for the file can be found in the kubernetes-template repository[2].
A description for the file is in the steering-committee docs[3], you might need
to search that page for "Security Contacts".

Please feel free to ping me on the PR when you make it, otherwise I will see when
you close this issue. :)

Thanks so much, let me know if you have any questions.

(This issue was generated from a tool, apologies for any weirdness.)

[1] https://groups.google.com/forum/#!topic/kubernetes-dev/codeiIoQ6QE
[2] https://github.com/kubernetes/kubernetes-template-project/blob/master/SECURITY_CONTACTS
[3] https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance-template-short.md

### PR DESCRIPTION
**Pull request by swatisehgal**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Bootstraps noderesourcetopology-api repo and populates mandatory files.

#### Which issue(s) this PR fixes:

Fixes: https://github.com/kubernetes/noderesourcetopology-api/issues/1 and https://github.com/kubernetes/community/issues/5738

#### Special notes for your reviewer:
This is an initial PR to bootstrap repo. This PR will be followed-up with a PR to populate the CRD API (update to https://github.com/kubernetes/kubernetes/pull/96275) with the goal to address https://github.com/kubernetes/community/issues/6308.

This PR is a prerequisite to https://github.com/kubernetes/kubernetes/pull/96275.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by swatisehgal** [comment]
Some of the labels got added due to an erroneous force push. Removing them.
/remove-area apiserver cloudprovider code-generation kubectl release-eng
/remove-sig architecture auth cli cloud-provider cluster-lifecycle instrumentation release storage

**Comment by dims** [comment]
- pull-kubernetes-dependencies : is failing because we updated `github.com/emicklei/go-restful/v3` in master branch, so something broke in a merge?
- pull-kubernetes-verify : may be add a doc.go and see if it works, you can run the 3 scripts by hand locally ( hack/verify-golangci-lint.sh, hack/verify-import-boss.sh, hack/verify-publishing-bot.py )
- pull-publishing-bot-validate : need to update `staging/publishing/rules.yaml` to look like one of the others (see cri-api for example?)

**Comment by swatisehgal** [comment]
Thanks @dims. Appreciate you input on this.

> * pull-kubernetes-dependencies : is failing because we updated `github.com/emicklei/go-restful/v3` in master branch, so something broke in a merge?

Thanks for the pointer, looks like I didn't have the updated version of  `github.com/emicklei/go-restful/v3` and needed to rebase.

> * pull-kubernetes-verify : may be add a doc.go and see if it works, you can run the 3 scripts by hand locally ( hack/verify-golangci-lint.sh, hack/verify-import-boss.sh, hack/verify-publishing-bot.py )

Added doc.go (didn't know I had to add that, followed the staging README religiously and that did not capture that we have to add doc.go. I will submit a PR to update the [README](https://github.com/kubernetes/kubernetes/blob/master/staging/README.md)). It is looking better now, just the pull-publishing-bot-validate lane still fails (see below).
> * pull-publishing-bot-validate : need to update `staging/publishing/rules.yaml` to look like one of the others (see cri-api for example?)

I already have that here:
https://github.com/kubernetes/kubernetes/pull/110252/files#diff-f916effca44e004de65234c58fe2274b998f8736b886faa96e07852b33e1e3bb, still getting: `E0616 10:38:50.592904     155 main.go:46] Error : noderesourcetopology-api not found in branch master` Anything obvious that I am missing here? 🤔

**Comment by swatisehgal** [comment]
/test pull-publishing-bot-validate
Lets retest as the test passed locally for me and yet it fails in the CI.
```
$ ./hack/verify-publishing-bot.py 
api dependencies
        apimachinery
apiextensions-apiserver dependencies
        api
        apimachinery
        apiserver
        client-go
        code-generator
        component-base
apimachinery dependencies
apiserver dependencies
        api
        apimachinery
        client-go
        component-base
cli-runtime dependencies
        api
        apimachinery
        client-go
client-go dependencies
        api
        apimachinery
cloud-provider dependencies
        api
        apimachinery
        apiserver
        client-go
        component-base
        component-helpers
        controller-manager
cluster-bootstrap dependencies
        api
        apimachinery
code-generator dependencies
component-base dependencies
        api
        apimachinery
        client-go
component-helpers dependencies
        api
        apimachinery
        client-go
controller-manager dependencies
        api
        apimachinery
        apiserver
        client-go
        component-base
cri-api dependencies
csi-translation-lib dependencies
        api
        apimachinery
kube-aggregator dependencies
        api
        apimachinery
        apiserver
        client-go
        code-generator
        component-base
kube-controller-manager dependencies
        api
        apimachinery
        apiserver
        client-go
        cloud-provider
        component-base
        component-helpers
        controller-manager
kube-proxy dependencies
        api
        apimachinery
        client-go
        component-base
kube-scheduler dependencies
        api
        apimachinery
        client-go
        component-base
kubectl dependencies
        api
        apimachinery
        cli-runtime
        client-go
        code-generator
        component-base
        component-helpers
        metrics
kubelet dependencies
        api
        apimachinery
        client-go
        component-base
legacy-cloud-providers dependencies
        api
        apimachinery
        apiserver
        client-go
        cloud-provider
        component-base
        component-helpers
        controller-manager
        csi-translation-lib
        mount-utils
metrics dependencies
        api
        apimachinery
        client-go
        code-generator
mount-utils dependencies
noderesourcetopology-api dependencies
        code-generator
pod-security-admission dependencies
        api
        apimachinery
        apiserver
        client-go
        component-base
sample-apiserver dependencies
        api
        apimachinery
        apiserver
        client-go
        code-generator
        component-base
sample-cli-plugin dependencies
        api
        apimachinery
        cli-runtime
        client-go
sample-controller dependencies
        api
        apimachinery
        client-go
        code-generator
processing : code-generator
processing : apimachinery
processing : api
  found dependency apimachinery
processing : client-go
  found dependency apimachinery
  found dependency api
processing : component-base
  found dependency apimachinery
  found dependency client-go
  found dependency api
processing : component-helpers
  found dependency apimachinery
  found dependency client-go
  found dependency api
processing : apiserver
  found dependency apimachinery
  found dependency client-go
  found dependency component-base
  found dependency api
processing : kube-aggregator
  found dependency client-go
  found dependency apiserver
  found dependency component-base
  found dependency apimachinery
  found dependency code-generator
  found dependency api
processing : sample-apiserver
  found dependency client-go
  found dependency apiserver
  found dependency component-base
  found dependency apimachinery
  found dependency code-generator
  found dependency api
processing : sample-controller
  found dependency apimachinery
  found dependency client-go
  found dependency code-generator
  found dependency api
processing : apiextensions-apiserver
  found dependency client-go
  found dependency apiserver
  found dependency component-base
  found dependency apimachinery
  found dependency code-generator
  found dependency api
processing : metrics
  found dependency apimachinery
  found dependency client-go
  found dependency code-generator
  found dependency api
processing : cli-runtime
  found dependency apimachinery
  found dependency client-go
  found dependency api
processing : sample-cli-plugin
  found dependency apimachinery
  found dependency client-go
  found dependency cli-runtime
  found dependency api
processing : kube-proxy
  found dependency apimachinery
  found dependency client-go
  found dependency component-base
  found dependency api
processing : kubelet
  found dependency apimachinery
  found dependency client-go
  found dependency component-base
  found dependency api
processing : kube-scheduler
  found dependency apimachinery
  found dependency client-go
  found dependency component-base
  found dependency api
processing : controller-manager
  found dependency client-go
  found dependency apiserver
  found dependency component-base
  found dependency apimachinery
  found dependency api
processing : cloud-provider
  found dependency client-go
  found dependency apiserver
  found dependency component-base
  found dependency apimachinery
  found dependency component-helpers
  found dependency controller-manager
  found dependency api
processing : kube-controller-manager
  found dependency client-go
  found dependency apiserver
  found dependency component-base
  found dependency apimachinery
  found dependency component-helpers
  found dependency cloud-provider
  found dependency controller-manager
  found dependency api
processing : cluster-bootstrap
  found dependency apimachinery
  found dependency api
processing : csi-translation-lib
  found dependency apimachinery
  found dependency api
processing : mount-utils
processing : legacy-cloud-providers
  found dependency client-go
  found dependency apiserver
  found dependency component-base
  found dependency apimachinery
  found dependency component-helpers
  found dependency cloud-provider
  found dependency csi-translation-lib
  found dependency controller-manager
  found dependency mount-utils
  found dependency api
processing : cri-api
processing : kubectl
  found dependency client-go
  found dependency metrics
  found dependency cli-runtime
  found dependency component-base
  found dependency apimachinery
  found dependency component-helpers
  found dependency code-generator
  found dependency api
processing : pod-security-admission
  found dependency client-go
  found dependency apiserver
  found dependency component-base
  found dependency apimachinery
  found dependency api
processing : noderesourcetopology-api
  found dependency code-generator
Done.
```

**Comment by dims** [comment]
@swatisehgal i think i know why, however we don't need to deal with it so skipped it. i think the CI job is using "master" branch as-is (without your PR) and it is failing the check.

**Comment by swatisehgal** [comment]
@dims Ah yeah, that makes sense. Thanks. If you/other reviewers would like, I can separate the rules.yaml updates to a separate PR. That should make CI happy.

**Comment by dims** [comment]
@swatisehgal +1 to split. thanks!

**Comment by swatisehgal** [comment]
>+1 to split. thanks!

@dims I have created another PR for updating publishing bot rules: https://github.com/kubernetes/kubernetes/pull/110629 and based this PR on that change.

I will rebase this PR once https://github.com/kubernetes/kubernetes/pull/110629 is merged.

File: staging/src/k8s.io/noderesourcetopology-api/OWNERS
Comment (kad) [inline staging/src/k8s.io/noderesourcetopology-api/OWNERS:9]: I would suggest to have @marquiz  added here as well as approver and/or reviewer. It would be good to have NFD maintainer to closely following those.

File: staging/src/k8s.io/noderesourcetopology-api/SECURITY_CONTACTS
Comment (kad) [inline staging/src/k8s.io/noderesourcetopology-api/SECURITY_CONTACTS:14]: same here, suggesting to add @marquiz

**Review by kad — COMMENTED** [review]
Generally looks good.
Two comments:
1. about maintainer - would be good to also have NFD maintainers to be owners of that code.
2.  it would be good to have clear plan where we going to use that API in core components. (e.g. replace Node Status with references to these node resources objects, or communication between kubelet and runtimes about resource discovery). Otherwise, I think it might be easier to move existing repository from foreign to kubernetes-sigs, rather than dealing with complexity of staging/*

File: staging/src/k8s.io/noderesourcetopology-api/OWNERS
Comment (swatisehgal) [reply staging/src/k8s.io/noderesourcetopology-api/OWNERS:9]: Sure, will add @marquiz here.

File: staging/src/k8s.io/noderesourcetopology-api/SECURITY_CONTACTS
Comment (swatisehgal) [reply staging/src/k8s.io/noderesourcetopology-api/SECURITY_CONTACTS:14]: Sure, will do.

**Comment by swatisehgal** [comment]
The error we are getting is as follows:
```
Traceback (most recent call last):
  File "hack/make-rules/../../hack/verify-publishing-bot.py", line 126, in <module>
    sys.exit(main())
  File "hack/make-rules/../../hack/verify-publishing-bot.py", line 121, in main
    raise Exception("missing rules for %s" % ','.join(str(s) for s in items))
Exception: missing rules for noderesourcetopology-api
```
Publishing bot rules are in separate PR: https://github.com/kubernetes/kubernetes/pull/110629.

**Comment by dims** [comment]
@swatisehgal i was wrong about the separate PR for Publishing bot rules, let's just add it here please

**Comment by dims** [comment]
> [](/k8s-ci-robot) ** pull-publishing-bot-validate **

**Comment by dims** [comment]
> Just ack-ing this was discussed in Nov 15, 2022 sig-node call.
> 
> @dims are you able to help shephard this with @swatisehgal ?

yes of course. we did this dance just end of last week for a new staging repo for dynamic resource ( cc @pohly )

we essentially ignored the `pull-publishing-bot-validate` job and landed the PR (using `/skip`) when we were ready

**Comment by swatisehgal** [comment]
/skip pull-publishing-bot-validate

Same issue as has been mentioned [here](https://github.com/kubernetes/kubernetes/pull/111023#issuecomment-1312272720).

**Review by ffromani — COMMENTED** [review]
/lgtm
from sig-node perspective. Content looks good and consistent.
Full disclosure: I can't say about the deps. At glance I see no problems and the bot is not complaining.

**Comment by swatisehgal** [comment]
> /lgtm from sig-node perspective. Content looks good and consistent. Full disclosure: I can't say about the deps. At glance I see no problems and the bot is not complaining.

Thanks @fromanirh for the review.

@dims the PR is ready for review/approval. We got @derekwaynecarr ACK from SIG-node to proceed with populating this repo. Can you please take a look if all looks good to you from deps perspective?

**Comment by dims** [comment]
> @dims the PR is ready for review/approval. We got @derekwaynecarr ACK from SIG-node to proceed with populating this repo. Can you please take a look if all looks good to you from deps perspective?

LGTM / APPROVE from deps perspective as there are no changes. 

```
/logs/artifacts /home/prow/go/src/k8s.io/kubernetes
go: downloading github.com/kubernetes-sigs/depstat v0.7.0
go: downloading github.com/spf13/cobra v1.1.1
go: downloading github.com/spf13/pflag v1.0.5
/home/prow/go/src/k8s.io/kubernetes
HEAD is now at 0f1e71c6476 Merge commit 'f9ec001fb65344a4a55bfb141e87a18b3099dc32'
Switched to a new branch 'base'
Files /logs/artifacts/stats-base.json and /logs/artifacts/stats.json are identical
```

@swatisehgal let's get some `/approve` and `/lgtm` from folks, i will be the last one to do that (i have root-ish approve umbrella and i don't want to use it right away)

**Review by mrunalp — APPROVED** [review]

**Comment by swatisehgal** [comment]
/assign @derekwaynecarr
for architecture review/approval

/assign @dchen1107 
for staging review/approval

**Comment by dims** [comment]
/milestone v1.27

just to be clear this will land after we open master branch up for 1.27

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

File: staging/publishing/rules.yaml
Comment (thockin) [inline staging/publishing/rules.yaml:2016]: I don't know the rules.yaml well, but this "dir" seems wrong to me - please confrim @dims?

Comment (dims) [reply staging/publishing/rules.yaml:2016]: @thockin we already landed something similar last week (see `dynamic-resource-allocation`), if this new `destination` looks like that then we are good!

Comment (dims) [reply staging/publishing/rules.yaml:2016]: heading out for an errand, will let @swatisehgal confirm that it does look like the other one.

Comment (dims) [reply staging/publishing/rules.yaml:2016]: (also this will land when `master` reopens for 1.27)

File: staging/src/k8s.io/noderesourcetopology-api/manifests/cr-instance-example.yaml
Comment (Huang-Wei) [inline staging/src/k8s.io/noderesourcetopology-api/manifests/cr-instance-example.yaml:1]: Given the API is gonna be in-tree, I suppose this folder is no longer needed :)

Comment (Huang-Wei) [reply staging/src/k8s.io/noderesourcetopology-api/manifests/cr-instance-example.yaml:1]: Makes sense to me. Thanks for the explanation!

File: staging/src/k8s.io/noderesourcetopology-api/OWNERS
Comment (derekwaynecarr) [inline staging/src/k8s.io/noderesourcetopology-api/OWNERS:3]: is there anyone else that wants to come forward for approver in seeding this repo?

**Review by derekwaynecarr — APPROVED** [review]
/approve

per Nov 29 sig-node call.

File: staging/src/k8s.io/noderesourcetopology-api/OWNERS
Comment (swatisehgal) [reply staging/src/k8s.io/noderesourcetopology-api/OWNERS:3]: @marquiz expressed interest in being a reviewer. I think we are good to go.

**Comment by swatisehgal** [comment]
/unhold
As we have got all the required approval on this PR .

@dims @fromanirh Would either of you be able to add an LGTM as it got removed after the PR was rebased. Thanks!

**Comment by fromanirh** [comment]
/lgtm
changes since last LGTM seems all caused by a rebase, I can't find anything which invalidates the previous LGTM.

**Comment by swatisehgal** [comment]
The master branch has opened for 1.27 contributions so removing the hold.

/unhold

**Comment by bart0sh** [comment]
@swatisehgal looks like the failing test expects existing `noderesourcetopology-api` directory in the master branch:
```
E1212 17:05:05.219577     155 main.go:46] Error : noderesourcetopology-api not found in branch master
F1212 17:05:05.220635     155 main.go:48] Invalid rules file "/home/prow/go/src/k8s.io/kubernetes/staging/publishing/rules.yaml"
```

Is it a known issue?

**Comment by swatisehgal** [comment]
> @swatisehgal looks like the failing test expects existing `noderesourcetopology-api` directory in the master branch:
> 
> ```
> E1212 17:05:05.219577     155 main.go:46] Error : noderesourcetopology-api not found in branch master
> F1212 17:05:05.220635     155 main.go:48] Invalid rules file "/home/prow/go/src/k8s.io/kubernetes/staging/publishing/rules.yaml"
> ```
> 
> Is it a known issue?

Yes, this is a know issue and we have to skip this job as the CI is still validating against the old tree and not taking into consideration that noderesourcetopology-api is being added in this PR (ref: https://github.com/kubernetes/kubernetes/pull/110252#issuecomment-1315733474).

**Comment by swatisehgal** [comment]
@fromanirh @dims Sorry to bother you both again. I had to rebase the PR which removed the LGTM. Can either of you add that so we can move this forward.

**Comment by fromanirh** [comment]
/lgtm
changes since last LGTM (https://github.com/kubernetes/kubernetes/compare/41a681cacd4b22903496cb52f2ffe5b92bf596f8..c6b29a0b05fe1fccbdbf1e430943d5995bc25a3e) looks harmless to me

**Comment by fromanirh** [comment]
feel free to unhold anytime when ready to merge

**Comment by swatisehgal** [comment]
/unhold

/skip
as indicated in the comment: https://github.com/kubernetes/kubernetes/pull/110252#issuecomment-1315733474

**Pull request merged**
